openedx/core/djangoapps/credit/signals.py

```diff
@@ -70,3 +99,290 @@ def listen_for_grade_calculation(sender, username, grade_summary, course_key, de
                     api.set_credit_requirement_status(
                         username, course_id, 'grade', 'grade', status="failed", reason={}
                     )
+
+
+def tag_course_content_with_partition_scheme(course_key, partition_scheme):  # pylint: disable=invalid-name
+    """ Create user partitions with provided partition scheme and tag credit
+    blocks for the given course with those user partitions.
+
+    Course and its all block will be updated with their respective last
+    editor's `edited_by`.
+
+    Args:
+        course_key (CourseKey): Identifier for the course
+        partition_scheme (str): Name of the user partition scheme
+
+    """
+    # TODO: Refactor code (move code to proper files e.g., 'common/lib/xmodule/xmodule/partitions/partitions.py')
+    # TODO: Increase logging (add logging for errors and success)
+
+    # get user partition with provided partition scheme
+    user_partition = UserPartition.get_scheme(partition_scheme)
```

Comment:
this block seems incorrect as written, because [`get_scheme`](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/partitions/partitions.py#L113-L126) returns a partition scheme, not a partition.  Note also that the method won't return `None`, and will raise its own Exception if the scheme you ask for hasn't been installed/configured.